script/hassfest/requirements.py

```diff
@@ -28,6 +28,42 @@
 PIP_REGEX = re.compile(r"^(--.+\s)?([-_\.\w\d]+.*(?:==|>=|<=|~=|!=|<|>|===)?.*$)")
 PIP_VERSION_RANGE_SEPARATOR = re.compile(r"^(==|>=|<=|~=|!=|<|>|===)?(.*)$")
 
+PACKAGE_CHECK_VERSION_RANGE = {
+    "aiohttp": "SemVer",
+    "attrs": "CalVer",
+    "grpcio": "SemVer",
+    "pyjwt": "SemVer",
+    "pytz": "CalVer",
+    "typing_extensions": "SemVer",
+}
+FORBIDEN_PACKAGES = {"setuptools", "wheel"}
+FORBIDEN_PACKAGE_EXCEPTIONS = {
```

Comment:
I'm not sure if we should have an exception for the integration (eg. "zha") or for the dependency (eg. "colorzero")

Reply:
I'd keep it as narrow as possible so I think it's better to do the exception per package instead of integration

Reply:
> > I'd keep it as narrow as possible so I think it's better to do the exception per package instead of integration
> 
> Updated to use the package name I've also adjusted so that a warning is displayed for these exemptions (error if it's not in the exemptions list)

Moved to #145381